### PR TITLE
Add quality declaration documents

### PR DESCRIPTION
**Pull request by gbiggs**
This PR adds [quality declaration documents](https://www.ros.org/reps/rep-2004.html) and (where necessary) basic README files too all packages in `rmf_demos`.

The resource packages meet the requirements to be Quality Level 3. All source-code-containing packages are currently QL4.

### CODE REVIEW

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:46]: I guess this changed now?

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:56]: Still TODO, will link PR when this is done

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:66]: TODO, same as above

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:78]: I guess leftover from copying from another repo?

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:82]: TODO

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:163]: Support for Eloquent has been dropped

**Review by luca-della-vedova — CHANGES_REQUESTED** [review]
Note, the files are now in the wrong folders (i.e. `rmf_gazebo_plugins`). I just put the comments for one file but they usually apply to all files (i.e. missing github actions, DCO / GPG signing being enforced, typos due to copying the declaration from another repo)

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md
Comment (luca-della-vedova) [reply rmf_gazebo_plugins/QUALITY_DECLARATION.md:56]: Done in #11

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md
Comment (mxgrey) [reply rmf_gazebo_plugins/QUALITY_DECLARATION.md:83]: The package-specific README.md does link to the quality declaration.

**Comment by mxgrey** [comment]
I just noticed that the packages these quality declarations are targeting no longer exist as they were split up between a bunch of new packages. I'm working on refactoring the quality declarations into the new packages.

**Review by mxgrey — APPROVED** [review]

**Review by mxgrey — APPROVED** [review]

**Comment by mxgrey** [comment]
@luca-della-vedova, it seems that GitHub will need you to change your review to an approval before it's willing to merge.

**Review by luca-della-vedova — APPROVED** [review]
Let's do it! :tada:

**Pull request merged**
